src/AggregateFunctions/AggregateFunctionMinMaxAny.h

```diff
@@ -620,7 +685,7 @@ struct SingleValueDataString //-V730
 
     bool changeIfLess(const IColumn & column, size_t row_num, Arena * arena)
     {
-        if (!has() || assert_cast<const ColumnString &>(column).getDataAt(row_num) < getStringRef())
+        if (!has() || Compatibility::getDataAtWithTerminatingZero(assert_cast<const ColumnString &>(column), row_num) < getStringRef())
```

Comment:
I can't wait to refactor this whole API to something saner, ideally with a complete revamp with https://github.com/ClickHouse/ClickHouse/pull/40633